onnxruntime/core/providers/rocm/rocm_execution_provider.cc

```diff
@@ -0,0 +1,1802 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
Would it be worth pre-applying some of @KeDengMS's simplifications from https://github.com/microsoft/onnxruntime/pull/5415 to this fork of the code?

Reply:
yeah, will do it.